include/pybind11/pybind11.h

```diff
@@ -207,7 +214,8 @@ class cpp_function : public function {
         PYBIND11_DESCR_CONSTEXPR auto types = decltype(signature)::types();
 
         /* Register the function with Python from generic (non-templated) code */
-        initialize_generic(rec, signature.text, types.data(), sizeof...(Args));
+        // Pass on the ownership over the `unique_rec` to `initialize_generic`. `rec` stays valid.
```

Comment:
I'm not too happy about the sort-of-implicit contract of "you take ownership, but can't yet destroy the object". I understand why this is the case, it's just not what "pass on the ownership" really means.

Reply:
No, I agree, but 1) it's already what was happening before (it's just syntactically explicit now), and 2) I wouldn't immediately know what to do otherwise? Return the pointer from `initialize_generic`? That also feels reasonably ugly/non-intuitive.

Reply:
Right. I was just thinking out loud.